TextField.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  TextField.swift
+//  PortalView
+//
+//  Created by Juan Franco Caracciolo on 4/10/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+import Foundation
+
+public struct TextFieldProperties<MessageType> {
+    
+    public var text: String
+    public var textAfterLayout: String?
+    public var onEvents: OnTextFieldEvents<MessageType>
+    
+    fileprivate init(
+        text: String = "",
+        textAfterLayout: String? = .none,
+        onEvents: OnTextFieldEvents<MessageType> = OnTextFieldEvents() ) {
+        self.text = text
+        self.textAfterLayout = textAfterLayout
+        self.onEvents = onEvents
+    }
+    
+    public func map<NewMessageType>(_ transform: (MessageType) -> NewMessageType) -> TextFieldProperties<NewMessageType> {
+        return TextFieldProperties<NewMessageType>(
+            text: self.text,
+            textAfterLayout: self.textAfterLayout,
+            onEvents: self.onEvents.map(transform)
+        )
+    }
+    
+}
+
+public struct OnTextFieldEvents<MessageType> {
```

Comment:
A new type cannot be nested inside a generic type like TextFieldProperties. [Reference](https://forums.developer.apple.com/thread/12298)